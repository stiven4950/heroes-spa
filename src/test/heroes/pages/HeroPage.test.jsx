import { fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import { HeroPage } from "../../../heroes/pages";

const mockedUseNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUseNavigate,
}));

describe("Pruebas en <HeroPage />", () => {
    test('debe de mostrarse correctamente', () => {
        const { container } = render(
            <MemoryRouter initialEntries={[`/hero/dc-green`]}>
                <Routes>
                    <Route path="hero/:id" element={<HeroPage />} />
                </Routes>
            </MemoryRouter>
        )

        expect(container).toMatchSnapshot();

        const image = screen.getByRole("img");
        expect(image.src).toContain('/assets/heroes/dc-green.jpg');
        expect(image.alt).toBe('Green Lantern');
    });

    test('debe navegar a la pagina de marvel si el heroe no existe', () => {
        render(
            <MemoryRouter initialEntries={[`/hero/dc-no-exists`]}>
                <Routes>
                    <Route path="hero/:id" element={<HeroPage />} />
                    <Route path="marvel" element={<h1>Página Marvel</h1>} />
                </Routes>
            </MemoryRouter>
        )

        expect(screen.getByText("Página Marvel")).toBeTruthy();
    });

    test('debe de navegar a la pantalla anterior', () => {
        render(
            <MemoryRouter initialEntries={['/hero/dc-green']}>
                <Routes>
                    <Route path="hero/:id" element={<HeroPage />} />
                </Routes>
            </MemoryRouter>
        )

        const button = screen.getByRole("button");
        fireEvent.click(button);

        expect(mockedUseNavigate).toHaveBeenCalledWith(-1);
    });
});