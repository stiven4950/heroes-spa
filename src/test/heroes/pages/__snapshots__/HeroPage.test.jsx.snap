// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HeroPage /> debe de mostrarse correctamente 1`] = `
<div>
  <div
    class="row mt-5"
  >
    <div
      class="col-4"
    >
      <img
        alt="Green Lantern"
        class="img-thumbnail animate__animated animate__fadeInLeft"
        src="/assets/heroes/dc-green.jpg"
      />
    </div>
    <div
      class="col-8"
    >
      <h3>
        Green Lantern
      </h3>
      <ul
        class="list-group list-group-flush"
      >
        <li
          class="list-group-item"
        >
           
          <b>
            Alter ego:
          </b>
           
          Alan Scott
           
        </li>
        <li
          class="list-group-item"
        >
           
          <b>
            Publisher:
          </b>
           
          DC Comics
           
        </li>
        <li
          class="list-group-item"
        >
           
          <b>
            First appearance:
          </b>
           
          All-American Comics #16
           
        </li>
      </ul>
      <h5
        class="mt-3"
      >
         Characters 
      </h5>
      <p>
        Alan Scott, Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz
      </p>
      <button
        class="btn btn-outline-primary"
      >
        Regresar
      </button>
    </div>
  </div>
</div>
`;
